### ISSUE TITLE
Split off from `.../eos/utils` the physics models into separate subdirectory

### ISSUE THREAD

**Issue by cBobeth** [eos/eos#438]
Over the time too many different types of utils ended up in `.../eos/utils`, and maybe as a first step one could split off all files that concern the three main physics models
* SM
* CKM scan model
* Wilson scan model

into a separate subdirectory, for example `.../eos/models`

**Comment by mreboud** [comment]
I concur, purely mathematical tools should also belong to a separate sub-directory I think.

**Comment by dvandyk** [comment]
Does this work?

```
eos/
  maths/       <-- build first (from this list)
    complex.hh
    derivative{.cc,.hh,_TEST.cc}
    gsl-interface{.hh,_TEST.cc}
    integrate{.cc,.hh,-impl.hh,_TEST.cc}
    integrate-cubature{.cc,.hh}
    polylog{.cc,.hh,_TEST.cc}
    power_of{.hh,_TEST.cc}
    szego-polynomials{.cc,.hh,_TEST.cc} (once they arrive)
  utils/
    ...       <-- everything else?
  models/     <-- build last (from this list, prior to all observables)
    model{.cc,.hh}
    standard-model{.cc,.hh,_TEST.cc}
    ckm{.cc,.hh,_TEST.cc}
    wet{.cc,.hh,_TEST.cc}
    top-loops{.cc,.hh,_TEST.cc}
```

I think it does not quite yet. We may need something like
```
eos/
  utils/      <-- log, lock, parameters, density, mutable, etc.
  maths/      <-- can use utils/
  physics/    <-- can use utils/ and maths/, e.g., QCD, quantum numbers, etc.
  models/     <-- can use utils/, maths/, and physics/
  [... observables]
```

**Comment by cBobeth** [comment]
Probably you know best what `...eos/utils` currently contains. I am not sure what will go into the `...eos/physics`, but if it is only QCD and a few other things that belong to the SM, one might put it into `...eos/models` in the sense 'models = short-distance models'. If things get too messy in `...eos/models` one could think of subdividing it further into `QCD`,  `SM`, `CKM` etc. I have seen there is also something on LCDA's in `...eos/utils`, which might does not really fit into the classification of **short-distance models**, but rather something like **hadronic models** (in the sense of models of hadronic quantities). In the latter also form factors enter, so maybe for the time being the LCDA-related files can go to `...eos/form-factors`?

But mostly I would prefer not to have a `...eos/physics`, because it can mean everything and nothing. I prefer to have something more specific instead.

**Comment by dvandyk** [comment]
Indeed, no ``physics/`` for now. I think this is (mostly) fixed as of #444. Some additional spring cleaning of ``eos/utils/`` is being handled in #438.
